swagger.yaml

```diff
@@ -1,78 +1,356 @@
+#          required: false
+#          type: string
+#          format: email
+#
+#        - name: eppn
+#          in: query
+#          description: "The person's email, specifically [onidusername@oregonstate.edu]"
+#          required: false
+#          type: string
+#          format: email
+#
+#        - name: upn
+#          in: query
+#          description: "Same as EPPN in the long term..."
+#          required: false
+#          type: string
+#          format: email
+#
+#        - name: EPPNPrior
+#          in: body
+#          description: "Any prior email addresses the user may have"
+#          required: false
+#          schema:
+#            $ref: "#/definitions/EPPNPriorArray"
+#
+#      # Expected responses for this operation:
+#      responses:
+#        # Response code
+#        "200":
+#          description: "Successful response"
+#          # A schema describing your response object.
+#          # Use JSON Schema format
+#          schema:
+#            $ref: "#/definitions/Person"
+#        "404":
+#          description: "Not Found"
+#          schema:
+#            $ref: "#/definitions/Error"
+#        "500":
+#          description: "Internal Server Error"
+#          schema:
+#            $ref: "#/definitions/Error"
+#      security:
+#        - people_api_key: []
+#
+#  /people/{eppn}:
+#    get:
+#      tags:
+#        - people
+#      description: "Finds Person by eppn"
+#      operationId: getPersonByEppn
+#      produces:
+#        - application/json
+#      parameters:
+#        - name: eppn
+#          in: path
+#          description: "EPPN of person to be fetched"
+#          required: true
+#          type: string
+#          format: email
+#
+#      responses:
+#        "200":
+#          description: "Successful response"
+#          schema:
+#            $ref: "#/definitions/Person"
+#        "404":
+#          description: "Resource not found"
+#          schema:
+#            $ref: "#/definitions/Error"
+#        "500":
+#          description: "Internal Server Error"
+#          schema:
+#            $ref: "#/definitions/Error"
+#      security:
+#        - people_api_key: []
+
+  /directory:
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
-      responses:
-        200:
-          description: Object containing information
-          schema:
-            $ref: '#/definitions/Info'
-  /sample:
-    get:
-      summary: Get message
-      description: This is a sample resource that returns a short message.
+      tags:
+        - directory
+      operationId: getDirectoryEntries
+      description: "Get all entities in directory. Can be filtered by query parameters"
+      produces:
+        - application/json
+      parameters:
+        - name: Full Name
+          in: query
+          description: "Full name of person to be searched for."
+          required: false
+          type: string
+
+        - name: Surname
```

Comment:
we should update this to "Last Name" as we did elsewhere. also, we probably need a "First Name" parameter.